moveit_ros/planning_interface/test/serialize_msg.py

```diff
@@ -0,0 +1,113 @@
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Bjarne von Horn
+#
+
+from moveit_ros_planning_interface._moveit_planning_interface_test_serialize_msg_cpp_helper import ByteStringTestHelper
+from geometry_msgs.msg import Vector3
+import unittest
+
+try:
+    # Try Python 2.7 behaviour first
+    from StringIO import StringIO
+    py_version_maj = 2
+except ImportError:
+    # Use Python 3.x behaviour as fallback and choose the non-unicode version
+    from io import BytesIO as StringIO
+    py_version_maj = 3
+
+
+class PythonMsgSerializeTest(unittest.TestCase):
+    def setUp(self):
+        self.helper = ByteStringTestHelper()
+
+    def test_EmbeddedZeros(self):
+        self.assertTrue(self.helper.compareEmbeddedZeros(b"\xff\xef\x00\x10"))
+
+    def test_ByteStringFromPchar(self):
+        # ByteString(const char*) constructor
+        self.assertEqual(self.helper.getBytesPChar(), b"abcdef")
+
+    def test_ByteStringFromStdString(self):
+        # ByteString(const std::string&) constructor
+        self.assertEqual(self.helper.getBytesStdString(), b"\xff\xfe\x10\x00\x00")
+
+    def test_ByteStringDefaultCtor(self):
+        self.assertEqual(self.helper.getDefaultBytes(), b"")
+
+    def test_CxxTupleToPy(self):
+        # sending a tuple from C++ to Python
+        ans = self.helper.getTuple()
+        self.assertIsInstance(ans, tuple)
+        self.assertEqual(len(ans), 1)
+        self.assertEqual(ans[0], b"abcdef")
+
+    def test_PyTupleToCxx(self):
+        # sending a tuple from Python to C++
+        self.assertTrue(self.helper.compareTuple((b"mno",)))
+
+    def test_sendMessage(self):
+        tmp = StringIO()
+        Vector3(x=1.0, y=-2.0, z=0.25).serialize(tmp)
+        self.assertTrue(self.helper.compareVector(tmp.getvalue()))
+
+    def test_recieveMessage(self):
+        tmp = Vector3()
+        tmp.deserialize(self.helper.getVector())
+        self.assertEqual(tmp, Vector3(1.0, -2.0, 0.25))
+
+    def test_rejectInt(self):
+        with self.assertRaisesRegexp(Exception, "Python argument types in"):
+            self.helper.compareEmbeddedZeros(4711)
+
+    def test_rejectIntTuple(self):
+        with self.assertRaisesRegexp(Exception, "Python argument types in"):
+            self.helper.compareEmbeddedZeros((4711,))
+
+    def test_rejectUnicode(self):
+        with self.assertRaisesRegexp(Exception, "Python argument types in"):
+            self.helper.compareEmbeddedZeros(u'kdasd')
+
+    @unittest.skipIf(py_version_maj == 2, "does not trigger with python 2.7")
+    def test_rejectUnicodeTuple(self):
+        with self.assertRaisesRegexp(RuntimeError, "Underlying python object is not a Bytes/String instance"):
+            self.helper.compareVectorTuple((u'kdasd',))
+
+
+if __name__ == '__main__':
+    import rostest
+    PKGNAME = 'moveit_ros_planning_interface'
+    NODENAME = 'moveit_test_message_serialisation'
+    rostest.rosrun(PKGNAME, NODENAME, PythonMsgSerializeTest)
```

Comment:
Doesn't it suffice to call all unittests? Do you really require ROS functionality here?
```suggestion
    unittest.main()
```